### DISCUSSION TITLE
How to reorder uploaded files?

### DISCUSSION THREAD

**Discussion by denisgurev** [pocketbase/pocketbase#1614] [Q&A] [answered]
I have a collection with multiple files upload field but I can't seem to find how to reorder already uploaded files. Do I have to delete and reupload them in the correct order?

**Comment by ganigeorgiev** [comment, marked as answer]
With [v0.11.0+](https://github.com/pocketbase/pocketbase/releases/tag/v0.11.0) you should be able to "reorder" uploaded files by just changing the order of the filenames in the field array value:

```js
// old value: ["file1.png", "file2.png", "file3.png"]
await pb.collection("example").update("RECORD_ID", {
    "myFileField": ["file3.png", "file1.png", "file2.png"]
})
```

The same apply for multi-`relation` and multi-`select` fields. This also means that you can remove multiple files at once by skipping their file names from the array, eg.:

```js
// keep only file2.png, aka. delete file1.png and file3.png
await pb.collection("example").update("RECORD_ID", {
    "myFileField": ["file2.png"]
    
    // the above is equivalent to
    "myFileField-": ["file1.png", "file3.png"]
})
```

**Reply by denisgurev** [reply]
Wow, thank you. I don't know why I didn't thought of that.
